### ISSUE TITLE
JSON can only represent references to arrays or hashes

### ISSUE THREAD

**Issue by morrisonj** [SysPete/Dancer2-Debugger#2]
Hi Pete

I'm getting the error "**encountered CODE(0x36d6f58), but JSON can only represent references to arrays or hashes at /usr/local/share/perl/5.20.2/Dancer2/Debugger.pm line 248**" when I use Dancer2::Plugin::Database with Dancer2::Plugin::Debugger and make any sort of database-> type call. Dancer2::Plugin::Debugger works fine on its own. Perhaps its an issue with Dancer2::Plugin::Database or perhaps I'm just being dumb or made some kind of mistake (or both).

Here's some code that produces the error:

test.pm:

```
package test;
use Dancer2;
use Dancer2::Plugin::Database;
use Dancer2::Plugin::Debugger;

our $VERSION = '0.1';

get '/' => sub {

        # retrieve basket
        my $product = database->quick_select('tablename', { id => '1' });

        template 'index';
};

true;
```

app.psgi:

```
#!/usr/bin/env perl

use strict;
use warnings;
use FindBin;
use lib "$FindBin::Bin/../lib";

use Plack::Builder;

use Dancer2::Debugger;
my $debugger = Dancer2::Debugger->new;

use test;
my $app = test->to_app;

builder {
        $debugger->mount;
        mount '/' => builder {
                $debugger->enable;
                $app;
        }
};
```

config.yml:

```
appname: "test"
layout: "main"
charset: "UTF-8"
autopage: 1
template: "template_toolkit"
template_toolkit: "TT"
session: "YAML"
plugins:
    Database:
        driver: 'mysql'
        database: 'dbname'
        host: 'localhost'
        port: 3306
        username: 'dbusername'
        password: 'dbpassword'
        connection_check_threshold: 10
        dbi_params:
            RaiseError: 1
            AutoCommit: 1
            mysql_enable_utf8: 1
    Debugger:
        enabled: 1
engines:
    template:
        template_toolkit:
            start_tag: '<%'
            end_tag: '%>'
            ENCODING: 'utf8'
```

Really liking the plugin. It will be really useful if I could just figure out this issue.

**Comment by SysPete** [comment]
Exception is due to a coderef being added to Dancer2 settings which the debugger code doesn't handle since I never expected coderefs to appear from config.

Here is the Dancer2::Plugin::Database code that inserts the coderef:
https://github.com/bigpresh/Dancer-Plugin-Database/blob/master/Shared/lib/Dancer/Plugin/Database/Core.pm#L273-L277
This is an entirely reasonable and sane thing to do so I need to add a variable dumper to the Settings panel as already exists for TemplateVariables.

**Comment by SysPete** [comment]
So the fix is simpler than I expected...
https://metacpan.org/pod/JSON#allow_unknown

This doesn't help anyone who decides to use a serializer other than the default JSON::MaybeXS but I can't really help in that case.

New release out very soon.

**Comment by SysPete** [comment]
Released to CPAN as version 0.006.

@morrisonj Many thanks for taking the time to create this issue.

**Comment by morrisonj** [comment]
Thanks @SysPete ! 

Just upgraded and tested it and it's working great. Thanks for such a quick fix 👍
